docs/artifacts/includes/nuget/nuget-publish-endpoint.md

```diff
@@ -18,14 +18,78 @@ ms.date: 02/18/2022
 
-1. If this is the first time using Azure Artifacts with Nuget.exe, select **Get the tools** button and follow the instructions to install the prerequisites.
+1. When using a local development environment, it is recommended to have the [Azure Artifacts Credential Provider](https://github.com/microsoft/artifacts-credprovider) installed which provides authentication when pushing or downloading packages (see examples section).
 
     1. Download the [latest NuGet version](https://www.nuget.org/downloads).
-    1. Download and install the [Azure Artifacts Credential Provider](https://github.com/microsoft/artifacts-credprovider#azure-artifacts-credential-provider).
 
-1. Follow the instructions in the **Project setup** to add a config file. 
+    1. Follow the instructions in the **Project setup** to add a config file. 
+
+        :::image type="content" source="../../media/project-setup.png" alt-text="Project setup":::
+
+1. For Azure hosted builds that interact with Azure Artifacts feeds, you must use the [NuGet Authenticate](https://docs.microsoft.com/azure/devops/pipelines/tasks/package/nuget-authenticate) task for the authentication(see examples section).
+
+1. For Github actions builds that interact with Azure Artifacts feeds, you may use [Github setup dotnet action](https://github.com/actions/setup-dotnet) for the authentication(see examples section).
+
+1. For other hosted builds that interact with Azure Artifacts feeds, you may use [Azure PAT]([url](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=Windows)) saved in `nuget.config` file for the authentication(see examples section), but please exercise caution because it may leak your PAT as part of build environement log collection or source control checkout, artifact upload accidentally.
+
+## Examples
+
+- In the example below, pushing "MyPackage" version "5.0.2" to an Azure Artifacts feed from a local development environment, the API key "AZ" is only used as a placeholder. An Azure Artifacts feed still requires the [Azure Artifacts Credential Provider](https://github.com/microsoft/artifacts-credprovider) to be installed to properly authenticate.
+
+```dotnetcli
+nuget push MyPackage.5.0.2.nupkg -src https://pkgs.dev.azure.com/{organization}/{project}/_packaging/{feed}/nuget/v3/index.json AZ
+```
+
+- In the example below, pushing "MyPackage" version "1.1.5" to an Azure Artifacts feed from an Azure hosted build, the API key "AZ" is only used as a placeholder. An Azure Artifacts feed still requires your pipeline to use the [NuGet Authenticate task](https://docs.microsoft.com/azure/devops/pipelines/tasks/package/nuget-authenticate?view=azure-devops).
+
+  ```cli
+    - task: NuGetAuthenticate@1
+      inputs:
+        nuGetServiceConnections: MyServiceConnection_ExternalServer
+        
+    - script: |
+        dotnet build mypackage/mypackage.csproj -c Release
```

Comment:
nit: I'd just include the push example. 

It's a slippery slope once you add all the steps.

Reply:
Running `dotnet build`, `dotnet pack`  in the same script is not the best idea either as they are implicit. 

You could always just run `dotnet pack`, but I still think it isn't necessary. 

Curious what others think.

Reply:
I agree. Many developers will want restore, build, pack, and push in different steps, so that if something fails, it's much more obvious what step it failed at, before even reading the logs.

Also, I find it odd that build and pack is done with the `dotnet` cli, but then push with nuget.exe.  If you're going to build with the dotnet CLI, you might as well also push with it.

Honestly, I don't think there should be separate pages for nuget.exe and dotnet cli. I think there should just be a single "push from cli" page, and it should only use the dotnet cli.  Let's not encourage customers to use nuget.exe.

Reply:
Sure, I removed unnecessary `dotnet build` from all examples.

Reply:
> Honestly, I don't think there should be separate pages for nuget.exe and dotnet cli. I think there should just be a single "push from cli" page, and it should only use the dotnet cli. Let's not encourage customers to use nuget.exe.

If I remember correctly there're some edge cases like mono which still use nuget.exe, so keep it separate for now. Removing a page easier thing to do next time, so let's deal next time.